MBBSEmu/Program.cs

```diff
@@ -617,5 +653,49 @@ private void PasswordReset()
             _logger.Info("Sysop Password Reset!");
             _doResetPassword = false;
         }
+
+        /// <summary>
+        ///     Rebuilds the internal MajorBBS/WG Account Database (BBSUSR.DAT) using the current SQLite Database
+        ///     and users that are already created. This is useful if you have a corrupted or missing BBSUSR.DB.
+        ///
+        ///     The BBSUSR.DAT file (ACCDB) is used by MajorBBS/WG to store user accounts. It is referenced by several
+        ///     internal API calls and is required for the system to function properly. Because of this, we only store the
+        ///     bare minimum amount of inofrmation required for this file to exist and be valid. Full user account information
+        ///     is stored within the internal MBBSEmu SQLite Database.
+        ///
+        ///     This might seem a little confusing, but internally in the MajorBBS code this database is referenced
+        ///     as "ACCDB" but the actual file name is BBSUSR.DAT.
+        /// </summary>
+        private void RebuildAccDb()
+        {
+            _logger.Info("Rebuilding BBSUSR.DAT...");
+
+            //Get Internal MBBSEmu User Account Database
+            var acct = _serviceResolver.GetService<IAccountRepository>();
+            var accounts = acct.GetAccounts();
+
+            //Verify there are valid accounts in the MBBSEmu Accounts Database
+            if (!accounts.Any())
+            {
+                _logger.Error("No Accounts Found in MBBSEmu Database, skipping rebuild of BBSUSR.DAT");
+                _logger.Error("Please consider using the -DBRESET command line argument to reset the internal databases to their default state");
+                return;
+            }
+
+            //Get BBSUSR.DAT and clear out existing records
+            var _accountBtrieve = _serviceResolver.GetService<IGlobalCache>().Get<BtrieveFileProcessor>("ACCBB-PROCESSOR");
+            _accountBtrieve.DeleteAll();
+
+            //Insert each record into BBSUSR.DAT
+            foreach (var a in accounts) 
+                _accountBtrieve.Insert(new UserAccount(a.userName).Data, LogLevel.Error);
+
+            //Verify the Counts are Equal
+            if (accounts.Count() != _accountBtrieve.GetRecordCount())
+                _logger.Warn($"MBBSEmu Database Account Count ({accounts.Count()}) does not match BBSUSR.DAT Account Count ({_accountBtrieve.GetRecordCount()})");
```

Comment:
this likely isn't necessary since Insert should throw if unable to insert a record